processor/eval/ledger_for_evaluator_test.go

```diff
@@ -0,0 +1,597 @@
+	require.NoError(t, err)
+	defer ld.Close()
+
+	ret, err :=
+		ld.LookupWithoutRewards(map[basics.Address]struct{}{test.AccountB: {}})
+	require.NoError(t, err)
+
+	accountDataRet := ret[test.AccountB]
+	require.NotNil(t, accountDataRet)
+	assert.Equal(t, accountData, *accountDataRet)
+}
+
+func TestLedgerForEvaluatorAccountDataMissingAccount(t *testing.T) {
+	l := test.MakeTestLedger("ledger")
+	defer l.Close()
+	ld, err := indxLeder.MakeLedgerForEvaluator(l)
+	require.NoError(t, err)
+	defer ld.Close()
+
+	var addr basics.Address
+	_, err = rand.Read(addr[:])
+	ret, err :=
+		ld.LookupWithoutRewards(map[basics.Address]struct{}{addr: {}})
+	require.NoError(t, err)
+
+	accountDataRet := ret[addr]
+	assert.Nil(t, accountDataRet)
+}
+
+func TestLedgerForEvaluatorAsset(t *testing.T) {
+	l := test.MakeTestLedger("ledger")
+	defer l.Close()
+	pr, _ := block_processor.MakeProcessorWithLedger(l, nil)
+
+	txn0 := test.MakeAssetConfigTxn(0, 2, 0, false, "", "", "", test.AccountA)
+	txn1 := test.MakeAssetConfigTxn(0, 4, 0, false, "", "", "", test.AccountA)
+	txn2 := test.MakeAssetConfigTxn(0, 6, 0, false, "", "", "", test.AccountA)
+	txn3 := test.MakeAssetConfigTxn(0, 8, 0, false, "", "", "", test.AccountB)
+	txn4 := test.MakeAssetDestroyTxn(1, test.AccountA)
+
+	block, err := test.MakeBlockForTxns(test.MakeGenesisBlock().BlockHeader, &txn0, &txn1, &txn2, &txn3, &txn4)
+	assert.Nil(t, err)
+	rawBlock := rpcs.EncodedBlockCert{Block: block, Certificate: agreement.Certificate{}}
+	err = pr.Process(&rawBlock)
+	assert.Nil(t, err)
+
+	ld, err := indxLeder.MakeLedgerForEvaluator(l)
+	require.NoError(t, err)
+	defer ld.Close()
+
+	ret, err :=
+		ld.LookupResources(map[basics.Address]map[ledger.Creatable]struct{}{
+			test.AccountA: {
+				{Index: 1, Type: basics.AssetCreatable}: {},
+				{Index: 2, Type: basics.AssetCreatable}: {},
+				{Index: 3, Type: basics.AssetCreatable}: {},
+			},
+			test.AccountB: {
+				{Index: 4, Type: basics.AssetCreatable}: {},
+			},
+		})
+	require.NoError(t, err)
+
+	expected := map[basics.Address]map[ledger.Creatable]ledgercore.AccountResource{
+		test.AccountA: {
+			ledger.Creatable{Index: 1, Type: basics.AssetCreatable}: {},
+			ledger.Creatable{Index: 2, Type: basics.AssetCreatable}: {
+				AssetHolding: &basics.AssetHolding{
+					Amount: txn1.Txn.AssetParams.Total,
+					Frozen: txn1.Txn.AssetFrozen,
+				},
+				AssetParams: &basics.AssetParams{
+					Total:         txn1.Txn.AssetParams.Total,
+					Decimals:      txn1.Txn.AssetParams.Decimals,
+					DefaultFrozen: txn1.Txn.AssetParams.DefaultFrozen,
+					UnitName:      txn1.Txn.AssetParams.UnitName,
+					AssetName:     txn1.Txn.AssetParams.AssetName,
+					URL:           txn1.Txn.AssetParams.URL,
+					MetadataHash:  txn1.Txn.AssetParams.MetadataHash,
+					Manager:       txn1.Txn.AssetParams.Manager,
+					Reserve:       txn1.Txn.AssetParams.Reserve,
+					Freeze:        txn1.Txn.AssetParams.Freeze,
+					Clawback:      txn1.Txn.AssetParams.Clawback,
+				},
```

Comment:
It's hard to tell from the diff if these types are the same. Can you use:
```suggestion
				AssetParams: &txn1.Txn.AssetParams,
```